api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseDaoImpl.java

```diff
@@ -470,6 +474,28 @@ protected void handleLocationReference(Criteria criteria, ReferenceParam locatio
 		}
 	}
 	
+	protected Optional<Criterion> handleParentReference(ReferenceParam parentReference) {
+		if (parentReference != null) {
+			if (parentReference.getChain() != null) {
+				switch (parentReference.getChain()) {
+					case Location.SP_NAME:
+						return propertyLike("name", parentReference.getValue());
+					case Location.SP_ADDRESS_CITY:
+						return propertyLike("cityVillage", parentReference.getValue());
+					case Location.SP_ADDRESS_STATE:
+						return propertyLike("stateProvince", parentReference.getValue());
+					case Location.SP_ADDRESS_COUNTRY:
+						return propertyLike("country", parentReference.getValue());
+					case Location.SP_ADDRESS_POSTALCODE:
+						return propertyLike("postalCode", parentReference.getValue());
+					case "":
+						return Optional.of(eq("locationId", Integer.parseInt(parentReference.getValue())));
```

Comment:
We shouldn't expect a `locationId` passed in here. We should expect a `uuid` for the location instead. Generally speaking, except for the special case of concepts, we should only be dealing with UUIDs (concepts are special for historical reasons).